database/binlog/temporal_fields.go

```diff
@@ -201,22 +201,17 @@ type datetime2FieldDescriptor struct {
 // in sql-common/my_time.c) for encoding detail.
 func NewDateTime2FieldDescriptor(nullable NullableColumn, metadata []byte) (
 	fd FieldDescriptor,
-	reamining []byte,
+	remaining []byte,
 	err error) {
 
 	d := &datetime2FieldDescriptor{}
 
-	remaining, err := d.init(
+	remaining, err = d.init(
 		mysql_proto.FieldType_DATETIME2,
 		nullable,
 		5,
 		metadata)
-
-	if err != nil {
```

Comment:
let's keep this check around.  it's useful to return nil values in case someone forgot to check for err

Reply:
OK, updated.